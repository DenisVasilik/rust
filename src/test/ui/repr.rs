#[repr]
//~^ ERROR attribute must be of the form `#[repr(...)]`
//~| WARN `repr` attribute must have a hint
struct _A {}

#[repr = "B"]
//~^ ERROR attribute must be of the form `#[repr(...)]`
//~| WARN `repr` attribute isn't configurable with a literal
struct _B {}

#[repr = "C"]
//~^ ERROR attribute must be of the form `#[repr(...)]`
//~| WARN `repr` attribute isn't configurable with a literal
struct _C {}

#[repr(C)]
struct _D {}

fn main() {}
