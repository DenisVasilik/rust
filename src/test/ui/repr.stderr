error: attribute must be of the form `#[repr(...)]`
  --> $DIR/repr.rs:1:1
   |
LL | #[repr]
   | ^^^^^^^

error: attribute must be of the form `#[repr(...)]`
  --> $DIR/repr.rs:6:1
   |
LL | #[repr = "B"]
   | ^^^^^^^^^^^^^

error: attribute must be of the form `#[repr(...)]`
  --> $DIR/repr.rs:11:1
   |
LL | #[repr = "C"]
   | ^^^^^^^^^^^^^

warning: `repr` attribute must have a hint
  --> $DIR/repr.rs:1:1
   |
LL | #[repr]
   | ^^^^^^^ needs a hint
   |
   = note: #[warn(bad_repr)] on by default
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute isn't configurable with a literal
  --> $DIR/repr.rs:6:1
   |
LL | #[repr = "B"]
   | ^^^^^^^^^^^^^ needs a hint
   |
   = help: valid hints include `#[repr(C)]`, `#[repr(packed)]`, `#[repr(rust)]` and `#[repr(transparent)]`
   = note: for more information, visit <https://doc.rust-lang.org/reference/type-layout.html>

warning: `repr` attribute isn't configurable with a literal
  --> $DIR/repr.rs:11:1
   |
LL | #[repr = "C"]
   | ^^^^^^^^^^^^^ help: give `repr` a hint: `#[repr(C)]`

error: aborting due to 3 previous errors

